engine/currency_state_manager.go

```diff
@@ -0,0 +1,279 @@
+	close(c.shutdown)
+	c.wg.Wait()
+	c.shutdown = make(chan struct{})
+	log.Debugf(log.ExchangeSys, "%s %s", CurrencyStateManagementName, MsgSubSystemShutdown)
+	atomic.StoreInt32(&c.started, 0)
+	return nil
+}
+
+// IsRunning safely checks whether the subsystem is running
+func (c *CurrencyStateManager) IsRunning() bool {
+	if c == nil {
+		return false
+	}
+	return atomic.LoadInt32(&c.started) == 1
+}
+
+func (c *CurrencyStateManager) monitor() {
+	defer c.wg.Done()
+	timer := time.NewTimer(0) // Prime firing of channel for initial sync.
+	for {
+		select {
+		case <-c.shutdown:
+			return
+		case <-timer.C:
+			var wg sync.WaitGroup
+			exchs, err := c.GetExchanges()
+			if err != nil {
+				log.Errorf(log.Global,
+					"%s failed to get exchanges error: %v",
+					CurrencyStateManagementName,
+					err)
+			}
+			for x := range exchs {
+				wg.Add(1)
+				go c.update(exchs[x], &wg, exchs[x].GetAssetTypes(true))
+			}
+			wg.Wait() // This causes some variability in the timer due to
+			// longest length of request time. Can do time.Ticker but don't
+			// want routines to stack behind, this is more uniform.
+			timer.Reset(c.sleep)
+		}
+	}
+}
+
+func (c *CurrencyStateManager) update(exch exchange.IBotExchange, wg *sync.WaitGroup, enabledAssets asset.Items) {
+	defer wg.Done()
+	for y := range enabledAssets {
+		err := exch.UpdateCurrencyStates(context.TODO(), enabledAssets[y])
+		if err != nil {
+			if errors.Is(err, common.ErrNotYetImplemented) {
+				// Deploy default values for outbound gRPC aspects.
+				var pairs currency.Pairs
+				pairs, err = exch.GetAvailablePairs(enabledAssets[y])
+				if err != nil {
+					log.Errorf(log.ExchangeSys, "%s %s %s: %v",
+						CurrencyStateManagementName,
+						exch.GetName(),
+						enabledAssets[y],
+						err)
+					return
+				}
+
+				// Deploys a full spectrum supported list for the currency states
+				update := map[currency.Code]currencystate.Options{}
+				for x := range pairs {
+					update[pairs[x].Base] = currencystate.Options{}
+					update[pairs[x].Quote] = currencystate.Options{}
+				}
+
+				b := exch.GetBase()
+				if b == nil {
+					log.Errorf(log.ExchangeSys, "%s %s %s: %v",
+						CurrencyStateManagementName,
+						exch.GetName(),
+						enabledAssets[y],
+						"cannot update because base is nil")
+					return
+				}
+				err = b.States.UpdateAll(enabledAssets[y], update)
+				if err != nil {
+					log.Errorf(log.ExchangeSys, "%s %s %s: %v",
+						CurrencyStateManagementName,
+						exch.GetName(),
+						enabledAssets[y],
+						err)
+				}
+				return
+			}
+			log.Errorf(log.ExchangeSys, "%s %s %s: %v",
+				CurrencyStateManagementName,
+				exch.GetName(),
+				enabledAssets[y],
+				err)
+		}
+	}
+}
+
+// CurrencyStateGetAllRPC returns a full snapshot of currency states, whether they
+// are able to be withdrawn, deposited or traded on an exchange for RPC.
+func (c *CurrencyStateManager) CurrencyStateGetAllRPC(exchName string) (*gctrpc.CurrencyStateResponse, error) {
```

Comment:
Given that we are in the CurrencyStateManager, I do not think you need to specify `CurrencyState` on these function names. Also update comments across each of these